leetcode/stringsLinearList/726NumberofAtoms.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+#include <string>
+#include <map>
+#include <iostream>
+#include <sstream>
+using namespace std;
+
+class Solution {
+private:
+    map<string, int> processParentheses(const string& formula, int startIndex) {
+        // startIndex is (
+        map<string, int> elementsInside;
+        int currentIndex = startIndex+1;
+        string currentAtom;
+        int currentAtomCount = 0;
+        while(currentIndex < formula.size()) {
+            char ch = formula[currentIndex];
+            if (ch == ')') {
```

Comment:
logic seems to be complex, using state machine?

Reply:
Overkill for a leetcode answer